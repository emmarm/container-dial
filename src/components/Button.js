import React from 'react';
import PropTypes from 'prop-types';
import styled from 'react-emotion';

import Icon from './Icon';

const Btn = styled('button')(
  {
    alignItems: 'center',
    border: 'none',
    borderRadius: 4,
    boxShadow:
      'rgba(50, 50, 93, 0.1) 0px 7px 14px, rgba(0, 0, 0, 0.08) 0px 3px 6px',
    color: 'white',
    cursor: 'pointer',
    display: 'flex',
    fontSize: 20,
    fontWeight: 100,
    height: 50,
    justifyContent: 'center',
    padding: '0 25px',
    ':disabled': {
      backgroundColor: '#777',
      cursor: 'auto'
    }
  },
  ({ danger, narrow, padded, primary, theme }) => ({
    backgroundColor: primary ? theme.primary : danger ? '#FF3762' : '#bbb',
    padding: narrow && '0 15px',
    paddingLeft: padded && 30,
    ':hover': {
      backgroundColor: primary ? theme.light : danger ? '#FF5F81' : '#c8c8c8',
      ':disabled': {
        backgroundColor: '#777'
      }
    }
  })
);

const Button = ({
  className,
  danger,
  disabled,
  icon,
  narrow,
  onClick,
  primary,
  submit,
  text
}) => (
  <Btn
    className={className}
    danger={danger}
    disabled={disabled}
    onClick={onClick}
    narrow={narrow}
    padded={!!icon && !!text}
    primary={primary}
    type={submit ? 'submit' : 'button'}
  >
    {text}
    {icon && <Icon icon={icon} padded={!!text} />}
  </Btn>
);

Button.propTypes = {
  className: PropTypes.string,
  danger: PropTypes.bool,
  disabled: PropTypes.bool,
  icon: PropTypes.string,
  onClick: PropTypes.func,
  primary: PropTypes.bool,
  submit: PropTypes.bool,
  text: PropTypes.string
};

export default Button;
