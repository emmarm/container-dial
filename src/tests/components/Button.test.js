import React from 'react';
import { shallow } from 'enzyme';

import Button from '../../components/Button';

describe('Button component', () => {
  const props = {
    onClick: () => {},
    text: 'Click Me'
  };
  const wrapper = shallow(<Button {...props} />);

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('sets button text correctly', () => {
    const buttonText = wrapper
      .find('Styled(button)')
      .dive('button')
      .text();
    expect(buttonText).toBe(props.text);
  });

  describe('Button component with boolean props', () => {
    it('sets danger, disabled, narrow, and primary to true when passed in props', () => {
      const wrapperWithProps = shallow(
        <Button {...props} danger disabled narrow primary />
      );
      expect(wrapperWithProps.prop('danger')).toBeTruthy;
      expect(wrapperWithProps.prop('disabled')).toBeTruthy;
      expect(wrapperWithProps.prop('narrow')).toBeTruthy;
      expect(wrapperWithProps.prop('primary')).toBeTruthy;
    });
  });

  describe('Button component with icon', () => {
    const icon = 'image.jpg';
    const wrapperWithIcon = shallow(<Button {...props} icon={icon} />);

    it('matches snapshot', () => {
      expect(wrapperWithIcon).toMatchSnapshot();
    });

    it('shows icon', () => {
      const buttonIcon = wrapperWithIcon.find('Icon');
      expect(buttonIcon.length).toBe(1);
      expect(buttonIcon.prop('icon')).toBe(icon);
    });

    it('sets padded to true when text and icon props', () => {
      expect(wrapperWithIcon.prop('padded')).toBeTruthy();
    });
  });
});
