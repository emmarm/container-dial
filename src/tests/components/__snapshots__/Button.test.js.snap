// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Button component with icon matches snapshot 1`] = `
<Styled(button)
  onClick={[Function]}
  padded={true}
  type="button"
>
  Click Me
  <Icon
    icon="image.jpg"
    padded={true}
  />
</Styled(button)>
`;

exports[`Button component matches snapshot 1`] = `
<Styled(button)
  onClick={[Function]}
  padded={false}
  type="button"
>
  Click Me
</Styled(button)>
`;
